src/components/timeRangePicker/types.ts

```diff
@@ -0,0 +1,18 @@
+export type DateOption = {
+  id: string; // Date.toDateString()
+  weekday: string;
+  day: number;
+  month: number;
+  year: number;
+};
+
+export type TimeRangeOption = {
+  id: string;
+  start: string; // format: "hour:minute"
+  end: string; // format: "hour:minute"
```

Comment:
It's possible to force particular format of a string, check

https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0MwAUVWDEQC0gADVHBQGNJ3KAALygABEAAkYNAMOB4GRonzFBTBPB3AERcwABS0+mM9ysygc9wASmCKhwzQ8ZC8iHFzmB7BoXzcZmY7AYzEg-Fg0HUiQ58D0Ii8fLpDKZgj5SWxfPADlQAHJhAA5SASPlBFQAeS+ZHegmdWkgR1QjgUrmkeFATjNOmGWH0KAQiGhwkuNok4uiIgMHGxCyYrA4PCCJUUqHa3ihkEedsgyeoDS8vIA3ooSlT8-krqhKJxmABuUAqZLixB1Fd8zDt-KIDJWyBSkoo7grqnrrc7sB7sgH1BHk-4s8XhRSgC+m7BIORCgAK7Y8v07CINI3xASBKAALK-igvKcNBsFkEBA5DjgI5jhuE5To0ABMkHzoueZkB+oAAAYACSzuB+L-vkjHIehCj-rR267vuh78l+ZA-pxV4LuCPArgxTHtqx7EodxvEvvxNHHqe56iQB2Eyta+g0XhyAEU6k71I05HiZRy78iRAAM+QAIz2XyvAWRJd6CQwDgAeSjRUmgekkSuBmjuOJnTmRc6uUu1FHrZDlOS5JQ3pJHleYo-5AA

Reply:
@nowyDEV There would be some trouble when the hour or minute value is single-digital. 

`The specified value "9:46" does not conform to the required format.  The format is "HH:mm", "HH:mm:ss" or "HH:mm:ss.SSS" where HH is 00-23, mm is 00-59, ss is 00-59, and SSS is 000-999.`

In this case, I guess the type of Hour and Minute should be `string`, right? And if so, is it worth the effort to force the format? What do you think?

Reply:
It would make sense when it would be provided as `prop` to the component. Considering it's defined internally we don't need to force it at all cost.